src/style-spec/validate/validate_fog.js

```diff
@@ -18,6 +18,10 @@ export default function validateFog(options) {
         return errors;
     }
 
+    if (fog.color && fog.color.a === 0) {
+        errors = errors.concat([new ValidationError('fog', fog, 'fog.color alpha must be nonzero.')]);
```

Comment:
Could this have some validation test added in https://github.com/mapbox/mapbox-gl-js/tree/main/test/unit/style-spec/fixture to lock this code path?

Reply:
Will add. Fixing render tests now.